### PR TITLE
Update kubeadm install doc for v1.8

### PR DESCRIPTION
**Pull request by luxas**
@kubernetes/sig-cluster-lifecycle-pr-reviews

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/5676)

### CODE REVIEW

File: docs/setup/independent/install-kubeadm.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ This page shows how to use install kubeadm.
 
 {% capture prerequisites %}
 
-* One or more machines running Ubuntu 16.04+, CentOS 7 or HypriotOS v1.0.1+
+* One or more machines running Ubuntu 16.04+, CentOS 7, RHEL 7, Fedora 25 (YMMV) or HypriotOS v1.0.1+
```
Comment (timothysc) [inline docs/setup/independent/install-kubeadm.md]: Fedora 25+

File: docs/setup/independent/install-kubeadm.md [hunk 2]
```diff
@@ -55,38 +55,50 @@ documentation for the plugins about what port(s) those need.
...
 On each of your machines, install Docker.
-Version 1.12 is recommended, but v1.10 and v1.11 are known to work as well.
-Versions 1.13 and 17.03+ have not yet been tested and verified by the Kubernetes node team.
-For installation instructions, see
-[Install Docker](https://docs.docker.com/engine/installation/).
+Version v1.12 is recommended, but v1.11, v1.13 or 17.03 are known to work as well.
+Versions 17.06+ _might work_, but have not yet been tested and verified by the Kubernetes node team.
 
-## Installing kubectl
+You can for instance use the following commands to install Docker on your system:
```
Comment (timothysc) [inline docs/setup/independent/install-kubeadm.md]: You can use the following..

File: docs/setup/independent/install-kubeadm.md [hunk 3]
```diff
@@ -95,10 +107,12 @@ For each machine:
   deb http://apt.kubernetes.io/ kubernetes-xenial main
   EOF
   apt-get update
-  apt-get install -y kubelet kubeadm
+  apt-get install -y kubelet kubeadm kubectl
```
Comment (kad) [inline docs/setup/independent/install-kubeadm.md]: wouldn't kubectl be brought in by deps ?

File: docs/setup/independent/install-kubeadm.md (hunk 1 above)
Comment (luxas) [reply docs/setup/independent/install-kubeadm.md]: fixed

File: docs/setup/independent/install-kubeadm.md (hunk 2 above)
Comment (luxas) [reply docs/setup/independent/install-kubeadm.md]: fixed

File: docs/setup/independent/install-kubeadm.md (hunk 3 above)
Comment (luxas) [reply docs/setup/independent/install-kubeadm.md]: Yes, but this makes it explicit to the user

**Comment by luxas** [comment]
> Maybe add note about disabling swap ? it's new thing for 1.8, so if people are upgrading, they might end-up with non-starting kubelet

@kad Good idea. Done.

I tested this on Debian 9 and it works flawlessly as well. The only earlier blocker was that 7 and 8 didn't enable the memory cgroup, and we didn't want to tell the user to reboot & mock with their kernel cmdline to install k8s. Now that's not needed anymore :tada:

File: docs/setup/independent/install-kubeadm.md [hunk 4]
```diff
@@ -55,73 +56,105 @@ documentation for the plugins about what port(s) those need.
...
+yum install -y kubelet kubeadm
+systemctl enable kubelet && systemctl start kubelet
+```
 
 **Note:** Disabling SELinux by running `setenforce 0` is required to allow
-containers to access the host filesystem, which is required by pod networks for
-example. You have to do this until SELinux support is improved in the kubelet.
+containers to access the host filesystem, which is required for the `kubeadm init`
+process to complete successfully. You have to do this until SELinux support is
+improved in the kubelet.
```
Comment (timothysc) [inline docs/setup/independent/install-kubeadm.md]: is improved.

File: docs/setup/independent/install-kubeadm.md [hunk 5]
```diff
@@ -10,11 +10,12 @@ This page shows how to use install kubeadm.
 
 {% capture prerequisites %}
 
-* One or more machines running Ubuntu 16.04+, CentOS 7 or HypriotOS v1.0.1+
+* One or more machines running Ubuntu 16.04+, Debian 9, CentOS 7, RHEL 7, Fedora 25/26 (YMMV) or HypriotOS v1.0.1+
```
Comment (pipejakob) [inline docs/setup/independent/install-kubeadm.md]: nit: Based on the tone of this doc, I would expand YMMV.

File: docs/setup/independent/install-kubeadm.md [hunk 6]
```diff
@@ -55,73 +56,109 @@ documentation for the plugins about what port(s) those need.
...
-  gpgcheck=1
-  repo_gpgcheck=1
-  gpgkey=https://packages.cloud.google.com/yum/doc/yum-key.gpg
-          https://packages.cloud.google.com/yum/doc/rpm-package-key.gpg
-  EOF
-  setenforce 0
-  yum install -y kubelet kubeadm
-  systemctl enable kubelet && systemctl start kubelet
-  ```
+Please proceed with executing the following commands based on your OS as the `root` user by SSH-ing
```
Comment (pipejakob) [inline docs/setup/independent/install-kubeadm.md]: > You may for example become `root` by executing `sudo -i`

I hope we have a savvy audience, but maybe mention "by executing `sudo -i` after SSH-ing to each host."

File: docs/setup/independent/install-kubeadm.md [hunk 7]
```diff
@@ -55,73 +56,109 @@ documentation for the plugins about what port(s) those need.
...
+yum install -y kubelet kubeadm
+systemctl enable kubelet && systemctl start kubelet
+```
 
 **Note:** Disabling SELinux by running `setenforce 0` is required to allow
-containers to access the host filesystem, which is required by pod networks for
-example. You have to do this until SELinux support is improved in the kubelet.
+containers to access the host filesystem, which is required for the `kubeadm init`
+process to complete successfully. You have to do this until SELinux support
+is improved.
```
Comment (pipejakob) [inline docs/setup/independent/install-kubeadm.md]: Would it be appropriate to link to the tracking issue here, for those who are interested? I'm not sure how often we do that from our user docs.

Comment (luxas) [reply docs/setup/independent/install-kubeadm.md]: I don't think right now is necessary. We haven't specced out a plan yet

File: docs/setup/independent/install-kubeadm.md (hunk 6 above)
Comment (luxas) [reply docs/setup/independent/install-kubeadm.md]: done

File: docs/setup/independent/install-kubeadm.md (hunk 5 above)
Comment (luxas) [reply docs/setup/independent/install-kubeadm.md]: done

**Comment by luxas** [comment]
Yay! Thanks!

**Pull request merged**